### PR TITLE
[semver:patch] add check on the HTTP_CODE returned by the curl

### PR DESCRIPTION
**Pull request by PPFerrari**
ref https://github.com/ClouDesire/support/issues/4906

### CODE REVIEW

File: src/scripts/utility_check_url.sh [hunk 1]
```diff
@@ -9,5 +9,9 @@ echo "Retries: ${CHECK_RETRIES}"
 echo "Sleep time: ${SLEEP_TIME}"
 echo "URL: ${CHECK_URL}"
 
-docker run --network "container:${CONTAINER_NAME}" --rm curlimages/curl:7.75.0 --retry "${CHECK_RETRIES}" --retry-delay ${SLEEP_TIME} --max-time 10 --retry-all-errors "${CHECK_URL}"
+HTTP_RESPONSE_CODE=$(docker run --network "container:${CONTAINER_NAME}" --rm curlimages/curl:7.75.0 -s -o /dev/null -w "%{http_code}" --retry "${CHECK_RETRIES}" --retry-delay ${SLEEP_TIME} --max-time 10 --retry-all-errors "${CHECK_URL}")
```
Comment (gionn) [inline src/scripts/utility_check_url.sh:12]: > -f, --fail
> 
> (HTTP) Fail silently (no output at all) on server errors. This is mostly done to enable scripts etc to better deal with failed attempts. In normal cases when an HTTP server fails to deliver a document, it returns an HTML document stating so (which often also describes why and more). This flag will prevent curl from outputting that and return error 22.
> 
> This method is not fail-safe and there are occasions where non-successful response codes will slip through, especially when authentication is involved (response codes 401 and 407). 

https://curl.se/docs/manpage.html

**Review by gionn — CHANGES_REQUESTED** [review]

File: src/scripts/utility_check_url.sh (hunk 1 above)
Comment (PPFerrari) [reply src/scripts/utility_check_url.sh:12]: lo aggiungo alle altre opzioni?

Comment (gionn) [reply src/scripts/utility_check_url.sh:12]: non serve che ti parsi a mano il response code con `-f`, anche perché così non stai gestendo i `20x` che sono comunque response valide

**Review by gionn — CHANGES_REQUESTED** [review]

File: src/scripts/utility_check_url.sh (hunk 1 above)
Comment (PPFerrari) [reply src/scripts/utility_check_url.sh:12]: quindi dovrei uscire con errore solo se ritorna `error code 22`? Funziona anche chiamando la `curl` da dentro un container?

Comment (gionn) [reply src/scripts/utility_check_url.sh]:
```
gionn at gt-g6 in ~/src/ops (master●)
$ docker run --rm "curlimages/curl:7.75.0" -f httpbin.org/status/500
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
curl: (22) The requested URL returned error: 500

gionn at gt-g6 in ~/src/ops (master●)
$ echo $?
22
```

File: src/scripts/utility_check_url.sh [hunk 2]
```diff
@@ -4,10 +4,24 @@ if [ "${CONTAINER_NAME}" == "" ]; then
...
+  -f \
+  --retry ${CHECK_RETRIES} \
+  --retry-delay ${SLEEP_TIME} \
+  --max-time 10 \
+  --retry-all-errors \
+  "${CHECK_URL}"
+
+if [ $? -ne 0 ]; then
+  exit 1
+fi
```
Comment (gionn) [inline src/scripts/utility_check_url.sh:26]: or simply 

```suggestion
exit $?
```

**Review by gionn — APPROVED** [review]

**Pull request merged**